demisto_sdk/commands/download/downloader.py

```diff
@@ -160,271 +141,316 @@ def __init__(
+
+                uuid_mapping = self.create_uuid_to_name_mapping(custom_content_objects=all_custom_content_objects)
+
+                # Filter custom content so that we'll process only downloaded content
+                downloaded_content_objects = self.filter_custom_content(
+                    custom_content_objects=all_custom_content_objects)
+
+                # Replace UUID IDs with names in filtered content (only content we download)
+                changed_uuids_count = 0
+                for _, file_object in downloaded_content_objects.items():
+                    if self.replace_uuid_ids(custom_content_object=file_object, uuid_mapping=uuid_mapping):
+                        changed_uuids_count += 1
+
+                if changed_uuids_count > 0:
+                    logger.debug(f"Replaced UUID IDs in {changed_uuids_count} custom content items.")
 
-        return 0
+            existing_pack_data = self.build_existing_pack_structure(existing_pack_path=Path(self.output_pack_path))
+
+            result = self.write_files_into_output_path(downloaded_content_objects=downloaded_content_objects,
+                                                       existing_pack_structure=existing_pack_data)
+
+            if result:
+                logger.info(f"Content items downloaded successfully into {self.output_pack_path}.")
+                return 0
+
+            logger.error(f"Download process failed.")
+            return 1
+
+        except Exception as e:
+            if not isinstance(e, HandledError):
+                logger.error(f"Error: {e}")
+
+            logger.debug(traceback.format_exc())
+            return 1
 
     def verify_flags(self) -> bool:
         """
-        Verifies that the flags configuration given by the user is correct
-        :return: The verification result
+        Verify that the flags provided by the user are valid and used correctly.
+
+        Returns:
+            bool: Whether the flags are valid or not.
         """
-        is_valid = True
-        if not self.list_files:
-            output_flag, input_flag = True, True
-            if not self.output_pack_path:
-                output_flag = False
-                logger.info("[red]Error: Missing option '-o' / '--output'.[/red]")
-            if not self.input_files:
-                if not self.all_custom_content and not self.regex:
-                    input_flag = False
-                    logger.info("[red]Error: Missing option '-i' / '--input'.[/red]")
-            if not input_flag or not output_flag:
-                is_valid = False
-
-        if self.download_system_item and not self.system_item_type:
-            logger.info(
-                "[red]Error: Missing option '-it' / '--item-type', "
-                "you should specify the system item type to download.[/red]"
-            )
-            is_valid = False
+        # If listing files, input / output flags are ignored.
+        if self.list_files:
+            return True
```

Comment:
log this, and any unexpected behaviors, as debug, please

Reply:
It's already logged here https://github.com/demisto/demisto-sdk/pull/3295/files#r1285236787
(before generating the list and printing it)

This method just prints errors if flags are missing / aren't used correctly.